nadabot/models.py

```diff
@@ -0,0 +1,187 @@
+    )
+    admins = models.ManyToManyField(
+        Account,
+        related_name="admin_registry",
+        help_text=_("registry admins."),
+    )
+    source_metadata = models.JSONField(
+        _("source metadata"),
+        null=False,
+        help_text=_("nadabot registry source metadata."),
+    )
+
+class Provider(models.Model):
+    id = models.PositiveIntegerField(
+        _("provider id"),
+        primary_key=True,
+        help_text=_("Provider id."),
+    )
+    contract_id = models.CharField(
+        _("contract ID"),
+        max_length=100,
+        null=False,
+        help_text=_("Contract ID of the external contract that is the source of this provider.")
+    )
+    method_name = models.CharField(
+        _("method name"),
+        max_length=100,
+        null=False,
+        help_text=_("Method name of the external contract that is the source of this provider.")
+    )
+    provider_name = models.CharField(
+        _("provider name"),
+        max_length=64,
+        null=False,
+        help_text=_("Name of the provider, e.g. 'I Am Human'.")
+    )
+    description = models.TextField(
+        _("description"),
+        blank=True,
+        null=True,
+        help_text=_("Description of the provider.")
+    )
+    status = models.CharField(
+        _("status"),
+        max_length=20,
+        choices=ProviderStatus,
+        null=False,
+        help_text=_("Status of the provider.")
+    )
+    admin_notes = models.TextField(
+        _("admin notes"),
+        blank=True,
+        null=True,
+        help_text=_("Admin notes, e.g. reason for flagging or marking inactive.")
+    )
+    default_weight = models.PositiveIntegerField(
+        _("default weight"),
+        null=False,
+        help_text=_("Default weight for this provider, e.g. 100.")
+    )
+    gas = models.BigIntegerField(
+        _("gas"),
+        blank=True,
+        null=True,
+        help_text=_("Custom gas amount required.")
+    )
+    tags = models.JSONField(
+        _("tags"),
+        blank=True,
+        null=True,
+        help_text=_("Optional tags.")
+    )
+    icon_url = models.URLField(
+        _("icon URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional icon URL.")
+    )
+    external_url = models.URLField(
+        _("external URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional external URL.")
+    )
+    submitted_by = models.CharField(
+        _("submitted by"),
+        max_length=100,
+        null=False,
+        help_text=_("User who submitted this provider.")
+    )
+    submitted_at_ms = models.DateTimeField(
+        _("submitted at (milliseconds)"),
+        null=False,
+        help_text=_("Timestamp of when this provider was submitted.")
+    )
+    stamp_count = models.BigIntegerField(
+        _("stamp count"),
+        default=0,
+        help_text=_("Total number of times this provider has been used successfully."),
+        db_index=True
+    )
```

Comment:
We shouldn't need this as a model field as we can access via `Provider.stamps.count()` and include in serializer if we want